File: openharmony-sig/ohos_react_native/tester/harmony/react_native_openharmony/src/main/cpp/RNOHCorePackage/TurboModules/DevLoadingViewTurboModule.cpp
/**
 * Copyright (c) 2024 Huawei Technologies Co., Ltd.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE-MIT file in the root directory of this source tree.
 */

#include "RNOHCorePackage/TurboModules/DevLoadingViewTurboModule.h"

using namespace facebook;

namespace rnoh {
DevLoadingViewTurboModule::DevLoadingViewTurboModule(
    const ArkTSTurboModule::Context ctx,
    const std::string name)
    : ArkTSTurboModule(ctx, name) {
  methodMap_ = {
      ARK_METHOD_METADATA(showMessage, 3),
      ARK_METHOD_METADATA(hide, 0),
  };
}
} // namespace rnoh

File: openharmony-sig/ohos_react_native/tester/harmony/react_native_openharmony/src/main/ets/RNOHCorePackage/turboModules/DevLoadingViewTurboModule.ts
/**
 * Copyright (c) 2024 Huawei Technologies Co., Ltd.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE-MIT file in the root directory of this source tree.
 */

import { TurboModule } from "../../RNOH/ts"

export class DevLoadingViewTurboModule extends TurboModule {
  static readonly NAME = "DevLoadingView"

  public showMessage(message: string, withColor?: number, withBackgroundColor?: number) {
    this.ctx.devToolsController.eventEmitter.emit("SHOW_DEV_LOADING_VIEW", this.ctx.rnInstance.getId(), message, withColor, withBackgroundColor)
  };

  public hide() {
    this.ctx.devToolsController.eventEmitter.emit("HIDE_DEV_LOADING_VIEW", this.ctx.rnInstance.getId())
  };
}